mesonbuild/dependencies/qt.py

```diff
@@ -103,16 +103,19 @@ def _get_modules_lib_suffix(version: str, info: 'MachineInfo', is_debug: bool) -
 
 
 class QtExtraFrameworkDependency(ExtraFrameworkDependency):
-    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any], language: T.Optional[str] = None):
+    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any], qvars: T.Dict[str, str], language: T.Optional[str] = None):
         super().__init__(name, env, kwargs, language=language)
         self.mod_name = name[2:]
+        self.qt_extra_include_directory = qvars['QT_INSTALL_HEADERS']
 
     def get_compile_args(self, with_private_headers: bool = False, qt_version: str = "0") -> T.List[str]:
         if self.found():
             mod_inc_dir = os.path.join(self.framework_path, 'Headers')
             args = ['-I' + mod_inc_dir]
             if with_private_headers:
                 args += ['-I' + dirname for dirname in _qt_get_private_includes(mod_inc_dir, self.mod_name, qt_version)]
+            if self.qt_extra_include_directory:
+                args += ['-I' + self.qt_extra_include_directory]
```

Comment:
Hmm, but pkg-config always returns this in --cflags, and the Qmake dependency class unconditionally adds it.

Does QT_INSTALL_HEADERS sometimes not exist for frameworks?

Reply:
Random aside: why is this function called get_compile_args, the other dependency types set up self.compile_args more directly -- and get_compile_args is part of the `class Dependency` API. Does this just "not matter" as it's never used in practice except internally to construct QmakeQtDependency?

aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

Reply:
I don't know if QT_INSTALL_HEADERS does or doesn't exist for all invocations of a framework.  Just trying to avoid a `-I` without any path.